hail/python/hail/matrixtable.py

```diff
@@ -539,6 +539,138 @@ class MatrixTable(ExprContainer):
     def _from_java(jmir):
         return MatrixTable(ir.JavaMatrix(jmir))
 
+    @staticmethod
+    @typecheck(
+        globals=nullable(dictof(str, anytype)),
+        rows=nullable(dictof(str, sequenceof(anytype))),
+        cols=nullable(dictof(str, sequenceof(anytype))),
+        entries=nullable(dictof(str, sequenceof(sequenceof(anytype)))),
+    )
+    def from_parts(
+        globals: Optional[Dict[str, Any]] = None,
+        rows: Optional[Dict[str, Iterable[Any]]] = None,
+        cols: Optional[Dict[str, Iterable[Any]]] = None,
+        entries: Optional[Dict[str, Iterable[Iterable[Any]]]] = None
+    ) -> 'MatrixTable':
+        """Create a `MatrixTable` from its component parts.
+
+        Example
+        -------
+        >>> mt = hl.MatrixTable.from_parts(
+        ...     globals={'hello':'world'},
+        ...     rows={'foo':[1, 2]},
+        ...     cols={'bar':[2, 3]},
+        ...     entries={'baz':[[1, 2],[3, 4]]}
+        ... )
+        >>> mt.describe()
+        ----------------------------------------
+        Global fields:
+            'hello': str
+        ----------------------------------------
+        Column fields:
+            'col_idx': int32
+            'bar': int32
+        ----------------------------------------
+        Row fields:
+            'foo': int32
+            'row_idx': int32
+        ----------------------------------------
+        Entry fields:
+            'baz': int32
+        ----------------------------------------
+        Column key: None
+        Row key: ['row_idx']
+        ----------------------------------------
+        >>> mt.row.show()
+        +-------+---------+
+        |   foo | row_idx |
```

Comment:
Don't have much to say on the implementation here, and I'll let Dan re-review, but I'm curious here why `row_idx` comes second? If this is coming from evaluating this, how come the order would be different than the cols table?

Reply:
Oh yeah that is weird. It's as if dict orders its keys alphabetically. Fixed in latest commit.

Reply:
I still get the following in the unit tests, though I'm not sure if there's anything I can do about that
```python
>>> mt.entries().show()
+---------+-----+---------+-----+-------+
| row_idx | foo | col_idx | bar |   baz |
+---------+-----+---------+-----+-------+
|   int32 | str |   int32 | str | int32 |
+---------+-----+---------+-----+-------+
|       0 | "a" |       0 | "c" |     1 |
|       0 | "a" |       1 | "d" |     2 |
|       1 | "b" |       0 | "c" |     3 |
|       1 | "b" |       1 | "d" |     4 |
+---------+-----+---------+-----+-------+
```

Reply:
Interestingly, a `hl.Struct` equality comparison is independent of the order of fields. @tpoterba, is this expected?

Reply:
Starting in Python 3.7 [dict order](https://docs.python.org/3/library/stdtypes.html#mapping-types-dict) is guaranteed to be insertion order, so maybe we can force that so the row/col keys come first? Not sure how that plays with hail structs

Reply:
I think field order doesn't matter for Hail structs in practice in Python.

Reply:
OK! So we're happy with the ordering then?

Reply:
The ordering looks right to me now.